readthedocs/projects/utils.py

```diff
@@ -42,3 +44,33 @@ def safe_write(filename, contents):
     with open(filename, 'w', encoding='utf-8', errors='ignore') as fh:
         fh.write(contents)
         fh.close()
+
+
+def _get_search_queries_from_queryset(queryset, sort=True):
+    """
+    Return list of search queries from queryset.
+
+    If ``sort`` is True, resulted list will be ordered in the descending order,
+    from most searched query to least search query.
+    Sample returned data: ['query1', 'query2', 'query3']
+    :param queryset: Queryset of the class SearchQuery
+    :type queryset: projects.querysets.RelatedProjectQuerySetBase
+    :param sort: If set to true, result will be sorted in descending order of most searched query
+    :type sort: bool
+    :returns: list of search queries
+    :rtype: list
+    """
+    count_data = (
+        queryset.values('query')
+        .annotate(count=Count('id'))
+    )
+
+    if sort:
+        # If sort is true, order the results by ``count``.
+        count_data = count_data.order_by('-count')
+
+    final_values = count_data.values_list('query', flat=True)
```

Comment:
Feels a bit weird to be doing a `values` and a `values_list` on the same queryset. I think this can probably be simplified.

Reply:
> to be doing a values and a values_list on the same queryset

I am not sure that I followed.

Reply:
Ohh.. I get it.
I think it's okay.
Last `values_list` is just to extract the data in list format without iterating over manually.